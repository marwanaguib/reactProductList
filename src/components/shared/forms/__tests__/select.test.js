import React from "react";
import { shallow } from "enzyme";

import Select from "../select";

const props = {
  func: jest.fn(),
  name: "test",
  elemKey: "department",
  options: [{ id: 1, name: "option1" }],
};

describe("select", () => {
  it("render select to match snapshot", () => {
    const componetWrapper = shallow(<Select {...props} />);
    expect(componetWrapper.find(`option`).at(1).text()).toBe(
      props.options[0].name
    );
    expect(componetWrapper).toMatchSnapshot();
  });
});
