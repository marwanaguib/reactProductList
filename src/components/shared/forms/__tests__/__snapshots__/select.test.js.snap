// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select render select to match snapshot 1`] = `
<styled.form>
  <select
    onChange={[Function]}
  >
    <option
      value=""
    >
      Select test
    </option>
    <option
      key="1"
      value={1}
    >
      option1
    </option>
  </select>
</styled.form>
`;
